inst/include/Rcpp/sugar/functions/strings/trimws.h

```diff
@@ -133,7 +133,8 @@ inline Vector<STRSXP> trimws(const Vector<STRSXP>& x, const char* which = "both"
 }
 
 inline Matrix<STRSXP> trimws(const Matrix<STRSXP>& x, const char* which = "both") {
-    R_xlen_t i = 0, nr = x.nrow(), nc = x.ncol(), sz = x.size();
+    R_xlen_t i = 0, sz = x.size();
+    int nr = x.nrow(), nc = x.ncol();
```

Comment:
I think that's the way it's implemented currently, see [R-ints](https://cran.r-project.org/doc/manuals/r-devel/R-ints.html#Large-matrices).